imageGenSD.py

```diff
@@ -363,12 +242,61 @@ async def generate(interaction: dis.Interaction,
         msg['data']['post']['seed']            = seed
         msg['data']['post']['cfg_scale']       = cfg_scale
         msg['data']['post']['sampler']         = sampler
-        
-        disLog.debug(f"Posting user job {msg} to the queue.") 
+
+        disLog.debug(f"Posting user job {msg} to the queue.")
         result = job_queue.Add(msg)
-    
+
     await interaction.response.send_message(f'{result}', ephemeral=True)
 
+@IGSD_client.tree.command()
+@dac.checks.has_permissions(use_application_commands=True)
+async def hello(interaction: dis.Interaction):
+    """A test echo command to verify basic discord functionality.
+
+       Input  : interaction - the interaction context from Discord.
+
+       Output : None.
+    """
+    await interaction.response.send_message(f'Hi, {interaction.user.mention}', ephemeral=True, delete_after=9.0)
+
+@IGSD_client.tree.command()
+@dac.checks.has_permissions(use_application_commands=True)
+async def listprofiles(interaction: dis.Interaction):
+    """Returns a pagenated list of profile names and IDs owned by the caller.
+
+        Input  : interaction - the interaction context from Discord.
+
+        Output : N/A.
+    """
+    disLog = log.getLogger('discord')
+    profiles = db_ifc.GetProfiles(interaction.user.id)
+
+    if not profiles:
+
+        embed = dis.Embed(title="Owned characters",
+                          description=f"User <@{interaction.user.id}> does not own any characters!  Use the `/roll` command to create one!",
+                          color=0xec1802)
+        await interaction.response.send_message(content=f"<@{interaction.user.id}>", embed=embed)
+
+    else:
+
+        async def GetPage(page: int):
```

Comment:
Do you really need to define this function at this level?

Reply:
(after testing) It appears to be yes, the Callable passed into the Menu Pagination object seems to lose 'self' if the function is instead defined in the parent.  Probably some async vodo that causes issues.  FWIW I borrowed heavily [from this post](https://stackoverflow.com/questions/76247812/how-to-create-pagination-embed-menu-in-discord-py) for the pagination implementation; it seems there just isn't a readily-available version of a paginated menu available for discord.py